File: app/cpp/console/code/GDatabase.cpp
//===============================================
#include "GDatabase.h"
//===============================================
GDatabase* GDatabase::m_instance = 0;
//===============================================
GDatabase::GDatabase() {

}
//===============================================
GDatabase::~GDatabase() {
    
}
//===============================================
GDatabase* GDatabase::Instance() {
    if(m_instance == 0) {
        m_instance = new GDatabase;
    }
    return m_instance;
}
//===============================================
void GDatabase::open() {
    printf("ouvrir la base de donnees\n");
}
//===============================================
void GDatabase::load() {
    printf("charger les donnees\n");
}
//===============================================
void GDatabase::update() {
    printf("mettre a jour les donnees\n");
}
//===============================================
void GDatabase::save() {
    printf("sauvegarder les donnees\n");
}
//===============================================

File: app/php/code/GDatabase.php
<?php
class GDatabase {
    //===============================================
    private static $instance = null;
    //===============================================
    public function __construct() {

    }
    //===============================================
    public static function Instance() {
        if(is_null(self::$instance)) {
            self::$instance = new GDatabase();  
        }
        return self::$instance;
    }
    //===============================================
    public function open() {
        echo sprintf("on ouvre la base de données<br/>\n");
    }
    //===============================================
    public function load() {
        echo sprintf("on charge la base de données<br/>\n");
    }
    //===============================================
    public function update() {
        echo sprintf("on modifie la base de données<br/>\n");
    }
    //===============================================
    public function save() {
        echo sprintf("on enregistre la base de données<br/>\n");
    }
    //===============================================
    public function close() {
        echo sprintf("on ferme la base de données<br/>\n");
    }
    //===============================================
}
?>
